greynoise/icon_greynoise/actions/community_lookup/action.py

```diff
@@ -0,0 +1,46 @@
+import insightconnect_plugin_runtime
+from .schema import CommunityLookupInput, CommunityLookupOutput, Input, Component
+
+# Custom imports below
+from greynoise import GreyNoise
+from greynoise.exceptions import RequestFailure
+from insightconnect_plugin_runtime.exceptions import PluginException
+import pendulum
+
+
+class CommunityLookup(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="community_lookup",
+            description=Component.DESCRIPTION,
+            input=CommunityLookupInput(),
+            output=CommunityLookupOutput(),
+        )
+
+    def run(self, params={}):
+        gn_client = GreyNoise(
+            api_server=self.connection.server,
+            api_key=self.connection.api_key,
+            integration_name=self.connection.user_agent,
+            offering="community",
+        )
+        try:
+            resp = gn_client.ip(params.get(Input.IP_ADDRESS))
+            if resp["noise"] or resp["riot"]:
+                resp["last_seen"] = pendulum.parse(resp["last_seen"]).to_rfc3339_string()
+
+        except RequestFailure as e:
+            raise PluginException(
+                cause="Received HTTP %d status code from GreyNoise. Verify your input and try again." % e.args[0],
+                assistance="If the issue persists please contact support.",
+                data=f"{e.args[0]}, {e.args[1]['message']}",
+            )
+        except ValueError as e:
```

Comment:
Are we 100% this is a 404? There's not a chance of just an odd return here?

Reply:
The "ValueError" should only process on bad input, meaning whatever was queried was not a valid input (and therefore was not found by the API).  There shouldn't be any other condition that will trigger this exception, the rest should fall into the RequestFailure.  If you'd like me to update this message I can make it a more generic response.